packages/itwinui-react/src/core/Tabs/Tabs.tsx

```diff
@@ -526,74 +610,236 @@ export const Tabs = (props: TabsProps) => {
+TabDescription.displayName = 'Tabs.TabDescription';
 
-      {actions && (
-        <Box className='iui-tabs-actions-wrapper'>
-          <Box className='iui-tabs-actions'>{actions}</Box>
-        </Box>
-      )}
+// ----------------------------------------------------------------------------
+// Tabs.Actions component
+
+const TabsActions = React.forwardRef((props, ref) => {
+  const { className, children, ...rest } = props;
+
+  const { type } = useSafeContext(TabsContext);
+
+  if (type !== 'pill') {
+    return (
+      <Box className={cx('iui-tabs-actions', className)} ref={ref} {...rest}>
+        {children}
+      </Box>
+    );
+  } else {
+    return <></>;
+  }
+}) as PolymorphicForwardRefComponent<'div'>;
+TabsActions.displayName = 'Tabs.Actions';
+
+// ----------------------------------------------------------------------------
+// Tabs.Panel component
+
+type TabsPanelOwnProps = {
+  /**
+   * Value used to associate the panel with a given tab.
+   */
+  value: string;
+};
+
+const TabsPanel = React.forwardRef((props, ref) => {
+  const { value, className, children, ...rest } = props;
+
+  const { activeValue } = useSafeContext(TabsContext);
+
+  if (value === activeValue) {
+    return (
+      <Box
+        className={cx('iui-tabs-content', className)}
+        role='tabpanel'
+        id={value}
+        ref={ref}
+        {...rest}
+      >
+        {children}
+      </Box>
+    );
+  } else {
+    return <></>;
+  }
+}) as PolymorphicForwardRefComponent<'div', TabsPanelOwnProps>;
+TabsPanel.displayName = 'Tabs.Panel';
+
+// ----------------------------------------------------------------------------
+// Tabs legacy component
+
+type TabsLegacyProps = {
+  /**
+   * Content displayed to the right/bottom of the horizontal/vertical tabs
+   *
+   * If `type = 'pill'`, `actions` is not applicable.
+   */
+  actions?: React.ReactNode[];
+  /**
+   * Elements shown for each tab.
+   * Recommended to pass an array of `Tab` components.
+   */
+  labels: React.ReactNode[];
+  /**
+   * Handler for activating a tab.
+   */
+  onTabSelected?: (index: number) => void;
+  /**
+   * Index of the active tab.
+   */
+  activeIndex?: number;
+  /**
+   * Control whether focusing tabs (using arrow keys) should automatically select them.
+   * Use 'manual' if tab panel content is not preloaded.
+   * @default 'auto'
+   */
+  focusActivationMode?: 'auto' | 'manual';
+  /**
+   * Color of the bar on the active tab.
+   * @default 'blue'
+   */
+  color?: 'blue' | 'green';
+  /**
+   * Custom CSS class name for tabs.
+   */
+  tabsClassName?: string;
+  /**
+   * Custom CSS class name for tab panel.
+   */
+  contentClassName?: string;
+  /**
+   * Custom CSS class name for the tabs wrapper.
+   */
+  wrapperClassName?: string;
+  /**
+   * Content inside the tab panel.
+   */
+  children?: React.ReactNode;
+} & TabsOrientationProps &
+  TabsOverflowProps;
+
+const TabsComponent = React.forwardRef((props, forwardedRef) => {
```

Comment:
should we deprecate old api? i believe it was failing a11y

Reply:
We did not deprecate other legacy APIs. But maybe we should, to tell users please use composition.

Reply:
Both ways sound good. Additionally, if we are not deprecating the old API, would be good to have an example of it on the docs site. https://github.com/iTwin/iTwinUI/pull/1548#discussion_r1350187225

Reply:
Created #1622 with docs changes in website